### ISSUE TITLE
Allow passing on of authToken to plex api instance

### ISSUE THREAD

**Issue by MikeOne** [phillipj/node-plex-api#21]
I'm looking at a scenario now where I need to connect to a shared (and remote) PMS. I do not have a login for this PMS, but because it is shared, I can connect to it using the token I can fetch through plex.tv.

Basically, I have a side process that fetches the token, however, I can't seem to pass that on to the plex-api. I would like it very much if, next to username/password, I can also pass 'token'. I have got it working locally, however, I'm not sure if this breaks something (as I'm now using it ad-hoc). It is a fairly simple change:

api.js

    function PlexAPI(options, deprecatedPort) {
    var opts = options || {};
    var hostname = typeof options === 'string' ? options : options.hostname;
    this.hostname = hostname;
    this.port = deprecatedPort || opts.port || PLEX_SERVER_PORT;
    this.username = opts.username;
    this.password = opts.password;
    this.authToken = opts.token;
    this.authenticator = opts.authenticator || this._credentialsAuthenticator();
    this.options = opts.options || {};
    this.options.identifier = this.options.identifier || uuid.v4();
    this.options.product = this.options.product || 'Node.js App';
    this.options.version = this.options.version || '1.0';
    this.options.device = this.options.device || os.platform();
    this.options.deviceName = this.options.deviceName || 'Node.js App';
    this.options.platform = this.options.platform || 'Node.js';
    this.options.platformVersion = this.options.platformVersion || process.version;

As you can see I juist added this.authToken = opts.token

This already seems to work, however, I'm not sure what happens if you pass an invalid token. My guess is auth will fail (401) and it will try to fetch a correct token using the username and password?

**Comment by phillipj** [comment]
I cant see any issues with allowing `authToken` to be passed with `options` as you have there.

ATM getting a 401 response when `authToken` is set, would quickly reject the request, due to these lines inside `PlexAPI.prototype._authenticate()`:
```javascript
if (this.authToken) {
  return deferred.reject(new Error('Permission denied even after attempted authentication :( Wrong username and/or password maybe?'));
}
```

I'm all ears to better solutions here if needed!

**Comment by MikeOne** [comment]
If the simple fix works (just make it an optional opts) - I'm fine with it?

**Comment by phillipj** [comment]
Alright! Would you like to open a PR for that?

**Comment by MikeOne** [comment]
Sure, but it I'll be my first so I have to find out how to best go about it :-)

**Comment by phillipj** [comment]
Awesome, dont hesitate to ask if you got any questions :smile:

**Comment by phillipj** [comment]
Linking this with PR #24.

**Comment by phillipj** [comment]
Landed in master with 644b5766934250
